data/preprocessor.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable no-use-before-define */
+
+const Preprocessor = (jsonRaw = {}) => {
+  const json = { ...jsonRaw }; // make a copy so we don't corrupt the input
```

Comment:
do we not need a deep copy?

Reply:
You're right :+1: I thought the spread operator did that for us, will fix!